src/BraketSimulator.jl

```diff
@@ -342,7 +342,7 @@ end
         
         todo_tasks_ch = Channel{Int}(ch->foreach(ix->put!(ch, ix), 1:n_tasks), n_tasks)
 
-        max_parallel_threads = max_parallel > 0 ? max_parallel : 32
+        max_parallel_threads = max_parallel > 0 ? max_parallel : min(32, Threads.nthreads())
```

Comment:
Why no just `Threads.nthreads()`?

Reply:
For the reason I used before, if you have 90 cores (let's say) and a batch of 2000 items you can spam the scheduler to death right now by having too many top level workers submitting simulations, since each simulation itself uses threads internally. In the long term we should improve some of the gate kernels to use more chunking so that they spawn fewer tasks and then we can remove this 32.